src/robocop/linter/rules/lengths.py

```diff
@@ -803,6 +844,104 @@ def count_keyword_calls(self, node):
         return calls
 
 
+class VariableNameLengthChecker(VisitorChecker):
+    """Checker for max length variable names."""
+
+    too_long_variable_name: TooLongVariableNameRule
+
+    set_variable_keywords = {
+        "setlocalvariable",
+        "settestvariable",
+        "settaskvariable",
+        "setsuitevariable",
+        "setglobalvariable",
+    }
+
+    def __init__(self) -> None:
+        super().__init__()
+        self.variables: dict[str, tuple] = {}
+
+    def visit_Node(self, node) -> None:  # noqa: N802
+        if not node.errors:
+            self.generic_visit(node)
+
+    def visit_Scope(self, node: Keyword | TestCase | VariableSection):  # noqa: N802
```

Comment:
Is visit_Scope actual visitor from Robot we're overwriting, or simply our function name? If so, we could just simply name it after one of the visitors in line 876 (like visit_Keyword)